### PR TITLE
Bump UI test dependency

### PR DESCRIPTION
**Pull request by oli-obk**
This gives us the new diff renderer as well as the ability to run tests without parallelism if we'd want to.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -40,9 +40,12 @@ libc = "0.2"
...
 [dev-dependencies]
 colored = "2"
-ui_test = "0.1"
+ui_test = "0.2"
 # Features chosen to match those required by env_logger, to avoid rebuilds
-regex = { version = "1.5.5", default-features = false, features = ["perf", "std"] }
+regex = { version = "1.5.5", default-features = false, features = [
+    "perf",
+    "std",
+] }
```
Comment (RalfJung) [inline Cargo.toml:48]: Is this some kind of auto-formatter? I don't think I like how it formats this.

Comment (oli-obk) [reply Cargo.toml:48]: ugh... I just wanted syntax highlighting and now it's doing this everywhere

File: tests/compiletest.rs [hunk 2]
```diff
@@ -76,45 +86,35 @@ fn run_tests(
...
 
-    let output_conflict_handling = match (env::var_os("MIRI_BLESS").is_some(), skip_ui_checks) {
+    config.output_conflict_handling = match (env::var_os("MIRI_BLESS").is_some(), skip_ui_checks) {
         (false, false) => OutputConflictHandling::Error,
         (true, false) => OutputConflictHandling::Bless,
         (false, true) => OutputConflictHandling::Ignore,
         (true, true) => panic!("cannot use MIRI_BLESS and MIRI_SKIP_UI_CHECKS at the same time"),
     };
 
     // Pass on all unknown arguments as filters.
```
Comment (RalfJung) [inline tests/compiletest.rs:101]:
```suggestion
    // Handle command-line arguments.
```

File: Cargo.toml (hunk 1 above)
Comment (RalfJung) [reply Cargo.toml:48]: format-on-save considered harmful. ;)

**Comment by RalfJung** [comment]
Also please update this to include https://github.com/oli-obk/ui_test/pull/9. :)

File: tests/compiletest.rs [hunk 3]
```diff
@@ -76,57 +86,43 @@ fn run_tests(
...
+    if with_dependencies && use_std {
+        config.dependencies_crate_manifest_path =
+            Some(Path::new("test_dependencies").join("Cargo.toml"));
+        config.dependency_builder.args = vec![
+            "run".into(),
+            "--manifest-path".into(),
+            "cargo-miri/Cargo.toml".into(),
+            "--".into(),
+            "miri".into(),
+            "run".into(),
```
Comment (RalfJung) [inline tests/compiletest.rs:123]:
```suggestion
            "run".into(), // There is no `cargo miri build` so we just use `cargo miri run`.
```

**Pull request merged**